projects/knora-ui/src/lib/viewer/values/interval-value/interval-value.component.ts

```diff
@@ -0,0 +1,115 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {BaseValueComponent} from "../base-value.component";
+import {CreateIntervalValue, ReadIntervalValue, UpdateIntervalValue} from "@knora/api";
+import {AbstractControl, FormBuilder, FormControl, FormGroup, ValidatorFn} from "@angular/forms";
+import {Subscription} from "rxjs";
+import {Interval} from "./interval-input/interval-input.component";
+
+@Component({
+  selector: 'kui-interval-value',
+  templateUrl: './interval-value.component.html',
+  styleUrls: ['./interval-value.component.scss']
+})
+export class IntervalValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+
+  @Input() displayValue?: ReadIntervalValue;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+
+  customValidators = [];
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder) {
+    super();
+  }
+
+  standardValidatorFunc: (val: any, comment: string, commentCtrl: FormControl) => ValidatorFn = (initValue: any, initComment: string, commentFormControl: FormControl): ValidatorFn => {
+    return (control: AbstractControl): { [key: string]: any } | null => {
+
+      const invalid = (initValue.start === control.value.start && initValue.end === control.value.end)
```

Comment:
@mdelez Wherever we deal with complex objects, the class needs to provide its custom validation method, overriding the standard method.